### ISSUE TITLE
Brown bread is erroring in recipe_permutations (in 0.F-dev)

### ISSUE THREAD

**Issue by actual-nh** [CleverRaven/Cataclysm-DDA#47874]
In the 0.F-dev branch, brown_bread is causing test errors in recipe_permutations (comensible_test.cpp). (It says the calories from the recipe need to be at least 1040 (currently 1031) and ideally 1300.)

# Steps To Reproduce

Run the tests on 0.F-dev; see [this build](https://github.com/CleverRaven/Cataclysm-DDA/pull/47872/checks?check_run_id=2034906590).

# Expected behavior

Ideally, no test errors.

# Screenshots

```
(~[slow] ~[.])=>	recipe_permutations
(~[slow] ~[.])=>	-------------------------------------------------------------------------------
(~[slow] ~[.])=>	/home/runner/work/Cataclysm-DDA/Cataclysm-DDA/tests/comestible_test.cpp:120
(~[slow] ~[.])=>	...............................................................................
(~[slow] ~[.])=>	
(~[slow] ~[.])=>	/home/runner/work/Cataclysm-DDA/Cataclysm-DDA/tests/comestible_test.cpp:157: FAILED:
(~[slow] ~[.])=>	  CHECK( lower_bound <= mystats.calories.avg() )
(~[slow] ~[.])=>	with expansion:
Error: (~[slow] ~[.])=>	  1040.0f <= 1031.0
(~[slow] ~[.])=>	
(~[slow] ~[.])=>	
(~[slow] ~[.])=>	
(~[slow] ~[.])=>	RecipeID: brown_bread, default is 1300 Calories,
(~[slow] ~[.])=>	Current recipe range: 1031-1031, Average 1031
(~[slow] ~[.])=>	Average recipe Calories must fall within this range, derived from default Calories: 1040-1560
```

# Versions and configuration

Github Build Matrix (GCC 7, Ubuntu, Tiles, CMake in this case).

# Additional context

Ping: @Xaleth?

**Comment by actual-nh** [comment]
Suggestion: Change the recipe to use, instead of 1 of bread_flour and 5 of flour, 3 of bread_flour and 3 of flour. (Which actually matches whole wheat flour substitution - half white and half whole wheat.) The two are different by 7 Cal.

Alternatively, to match a commercially-made bread, I suspect upping the molasses and adding brown sugar would work. (They up sweet - and salty - tastes because those make you notice bitter - as in burnt - taste less... plus most people like sugar and salt!)

**Comment by actual-nh** [comment]
Now, the question of why this wasn't seen before... I'm working on it.

**Comment by Xaleth** [comment]
The tests passed when I pushed the commits in the PR to fix this, and then when it got merged it's now showing errors? :thinking:

**Comment by actual-nh** [comment]
Yeah, I know. The thing is, the tests were run using a master as of something like January 23rd, and there have been changes since then. (This makes me conclude that I should probably more frequently "force update" or "rebase" or whatever my PRs, to make sure they still work with the current master.) I just spent a couple of hours scanning through the changes since then, and I really don't know.

I think it may be best to not just be within the caloric range the test will accept, but as close to the central target as possible - then, it will take something more severe to put the recipe out of the range (hopefully also messing up others and leading to the discovery that there's a bug).

**Comment by actual-nh** [comment]
I think that in terms of matching masses, a lot of water will get cooked out of the molasses, and it will probably wind up about the caloric density of brown sugar. The fourth or so ingredient of the B&B brown bread is actually dextrose - aka glucose - so putting in more sugar (sucrose is glucose + fructose, fruit sugar) should work.

I suggest adding 1 unit of sugar, and having the recipe have a choice of either 1 additional unit of molasses or 13 additional units of sugar. If my reading of the JSON files is correct, that will give the result a range that includes 1300.

**Comment by actual-nh** [comment]
Currently doing a compile locally (old, slow computer) so I can check if the above suggestion works; will PR if it does.

**Comment by actual-nh** [comment]
I don't know why this isn't showing up as being fixed by #47915 - possibly because that one is a PR for the 0.F-dev branch, not the master branch?

**Comment by actual-nh** [comment]
Fixed by #47915; thank you, ZhilkinSerg.
